Ts sdk snippets

Superpat opened the pull request:
Also improved the examples to include information on readSignals usage.

Superpat committed:
sdk/ts add warning about http1

Superpat committed:
sdk/ts improve readme example

and add snippets to go site

Superpat committed:
sdk/ts fix type and import errors

Superpat committed:
ts/sdk fix type error in node readme example

bencroker commented on site/static/code_snippets/getting_started/setup.tssnippet:
Shouldn’t lines 15-24 be sufficient? If you reference the other SDK code snippets, they don’t show handling of the entire request. Something more thorough should ideally exists in the SDK’s readme file and, if appropriate in the `/examples` folder.

Superpat replied:
You're not wrong, I fixed it.

Superpat committed:
sdk/ts improve snippets and examples

Slimmed down the snippets setup example and added it to the readme, then
moved the readme examples to their own files in a new examples folder to
make place for other examples.

Also fixed md links in readme.

bencroker commented:
Please rename the `ts` folders to `typescript` (and the snippets too, better to be consistent).

Superpat committed:
sdk/ts --unstable-sloppy-imports is no longer required

bencroker commented on site/static/code_snippets/getting_started/setup.tssnippet:
Most of the code in this file is error handling. Could we remove that from the snippet and use show it in the usage section of the readme only?

Superpat replied:
I think reading signals is part of getting started, but if you insist I will remove it. If I keep the readSignals example then I'd keep the error handling too.

bencroker replied:
Reading signals should be included, but can we only keep the `reader.success` check, remove the duplication of reporting the error, and remove the second conditional for the existence of `foo`?

Superpat replied:
Sure

bencroker replied:
Thanks, that looks much better now!

Superpat committed:
sdk/ts fixed some type / import errors

Superpat committed:
Rename sdk/ts to sdk/typescript

Superpat committed:
sdk/typescript simplify setup snippet

bencroker commented:
Looks good, ready for me to merge?

Superpat commented:
Yes !

bencroker commented:
Merged, thanks!

bencroker commented:
Now I’m running it locally, I’m seeing inconsistencies with the other code snippets. Can you try to mirror this as closely as possible in a new PR please?
https://github.com/starfederation/datastar/blob/develop/site/static/code_snippets/getting_started/setup.phpsnippet
